src/components/FilterableProductTable.js

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react';
+import SearchBar from './SearchBar';
+import ProductTable from './ProductTable';
+
+function FilterableProductTable(props) {
+  const [products, setProducts] = useState(props.products);
+
+  const searchProduct = (input) => {
+    let productCopy = [...products];
+
```

Comment:
I did a rewrite of line 11-33 using `if`, `else if` and `else` statements.

It's the same result in the end, I just thought maybe it would be good to reiterate that you have `else if` as an option.
```
if(input.search === '' && input.inStock) {
      setProducts(productCopy.filter((item) => item.stocked));
    }
    else if(input.search === '') {
      setProducts(props.products);
    }
    else if(input.inStock) {
      setProducts(
        productCopy.filter(
          (item) =>
            item.name.toUpperCase().includes(input.search.toUpperCase()) &&
            item.stocked
        )
      );
    }
    else {
      setProducts(
        productCopy.filter((item) =>
          item.name.toUpperCase().includes(input.search.toUpperCase())
        )
      );
    }